odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,497 @@
+        """Implementation of ``self(displacement)``.
+
+        Parameters
+        ----------
+        displacement : `ProductSpaceVector`
+            The displacement field used in the linearized deformation.
+        """
+
+        return linear_deform(self.template, displacement)
+
+    def derivative(self, displacement):
+        """Gateaux derivative of the operator in ``displacement``.
+
+        Parameters
+        ----------
+        displacement: `ProductSpaceVector`
+            The point that the Gateaux derivative need to be computed at.
+
+        Returns
+        -------
+        deriv_op : `Operator`
+            The derivative evaluated at ``displacement``
+        """
+        deriv_op = LinDeforFixedTempDeriv(self.template, displacement)
+        return deriv_op
+
+
+class LinDeforFixedTempDeriv(Operator):
+    """Derivative of the fixed template linearized deformation operator.
+
+    This operator computes the Gateaux derivative of the fixed
+    template linearized deformation operator for the fixed
+    template ``I`` at given displacement ``v``:
+
+        u --> grad(I)(Id + v).T u
+
+    where the ``u`` is a given vector field and
+    the ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain of the target.
+    In diffeomorphic sense, the domain of ``I`` is the same
+    as the target.
+    """
+
+    def __init__(self, template, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        template : `DiscreteLpVector`
+            Fixed template deformed by the vector field.
+        displacement: `ProductSpaceVector`
+            The point that the Gateaux derivative needs to be computed at.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> disp_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> vector_field = disp_field_space.element([[1, 1, 1, 2, 1]])
+        >>> op = LinDeforFixedTempDeriv(template, disp_field)
+        >>> op(vector_field)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 5.0, -5.0, -10.0, 0.0])
+        """
+        assert displacement[0] in template.space
+
+        Operator.__init__(self, displacement.space, template.space,
+                          linear=True)
+
+        self.template = template
+        self.displacement = displacement
+
+    def _call(self, vector_field):
+        """Implementation of ``self(vector_field)``.
+
+        Parameters
+        ----------
+        vector_field: `ProductSpaceVector`
+            The evaluation point, i.e. a displacement field, for the
+            deformed gradient of the template.
+        """
+        grad = odl.Gradient(self._range)
```

Comment:
Never access private variables of other classes unless needed. Simply go for the public one instead here

```
self.range
```